app/src/main/java/com/example/poste/activities/NewPostActivity.java

```diff
@@ -0,0 +1,61 @@
+package com.example.poste.activities;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.Toast;
+import android.util.Log;
+
+
+import com.example.poste.R;
+import com.example.poste.callbacks.UpdateCallback;
+import com.example.poste.models.Folder;
+import com.example.poste.models.User;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+public class NewPostActivity extends AppCompatActivity {
+
+    private String title;
+
+    private String link;
+
+    private String description;
+
+    private String tags;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_new_post);
+        if (User.getUser().getToken() == null || User.getUser().getToken().isEmpty()) {
+            // user needs to login
+            Log.d("NewPostActivity", "User needs to login");
+            Intent loginIntent = new Intent(this, LoginActivity.class);
+            loginIntent.putExtra("return", true);
```

Comment:
So once user logs in, the app finishes that activity, and since this activity never finished, it bring it back here? 
Am I correct? 

Only concern I have is that, the login (or maybe dashboard) would update the user data such as folder and posts. And in this control flow I don't really see the update happening, which might be why I am unable to view the dropdown folders I mentioned in a previous comment

Reply:
Once user logs back in, LoginActivity (having received this 'extra' of return) knows to return to NewPostActivity (instead of Dashboard) to finish allowing the user to create a post, correct.

You're correct about data updating -- I'll add data updating to LoginActivity in this case (since they won't hit dashboard first)

The dropdown folders aren't fully implemented yet. I considered that part of EditPostActivity... although now I guess this is a whole new activity (this branch evolved over time) so I should probably create it here. Then, it can be shared in EditPostActivity, if we so desire.

Reply:
I added data updating in the appropriate location in NewPostActivity to ensure data is retrieved before trying to do anything fancy